public/ServiceWorker/index.js

```diff
@@ -0,0 +1,29 @@
+self.addEventListener('install', event => {
+    event.waitUntil(
+        caches.open('cache-v1-pwa')
+            .then(cache => {
+                return cache.addAll([
+                    '/'
+                ])
+            })
+            .then(() => self.skipWaiting())
+    )
+})
+self.addEventListener('activate', event => {
```

Comment:
This code sets up an event listener for the activate event on the service worker self. When the service worker is activated, the callback function is executed.